### PR TITLE
[graphql] allow configurable max page size per cursor connection

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

Allow users to set a max page size per page of a connection, and enforce it as a max (throw error if first or last exceeds the page size).

Removing db_query_cost.rs because we do not use it anymore, and it has a dependency on the hardcoded DEFAULT_PAGE_SIZE

## Test Plan 

Existing tests + new ones to validate we return an error when the page limit is exceeded on client request.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by stefan-mysten — APPROVED** [review]

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs [hunk 1]
```diff
@@ -533,6 +536,33 @@ impl PgManager {
...
+                return Err(
+                    DbValidationError::PageSizeExceeded(l, self.limits.max_page_size).into(),
+                );
+            }
+        }
+
+        // TODO (wlmyng): even though we do not allow passing in both first and last,
+        // per the cursor connection specs, if both are provided, from the response,
+        // we need to take the first F from the left and then take the last L from the right.
+        Ok(first.or(last).unwrap_or(self.limits.max_page_size) as i64)
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:563]: We should not use the max page size as the default as well -- we can make the default page size a configurable limit as well, perhaps, but by setting things up this way, we're making the thing that is most expensive for us to serve the path of least resistance for users which is a bad set-up.

**Review by amnn — COMMENTED** [review]
Thanks @wlmyng !

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs (hunk 1 above)
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:563]: Ah yeah that's fair
Any qualms if I tack it on to the service config?

Comment (amnn) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:563]: (a) No qualms!
(b) How/why are you still awake? 😬

Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:563]: There are three kinds of emails that would pull me out of bed immediately

1. Correspondence from a prince about a large sum of money sitting in a bank that I have yet to claim
2. 75k points for business class flights to Seoul
2. PR reviews from @amnn

Comment (amnn) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:563]: I'm hoping that it's (2)

**Pull request merged**